src/Umbraco.Web/Templates/TemplateUtilities.cs

```diff
@@ -183,5 +186,57 @@ public static string ResolveMediaFromTextString(string text)
         internal static string RemoveMediaUrlsFromTextString(string text)
             // see comment in ResolveMediaFromTextString for group reference
             => ResolveImgPattern.Replace(text, "$1$3$4$5");
+
+        internal static string FindAndPersistPastedTempImages(string html, int mediaParentFolder, int userId, IMediaService mediaService, IContentTypeBaseServiceProvider contentTypeBaseServiceProvider)
+        {
+            // Find all img's that has data-tmpimg attribute
+            // Use HTML Agility Pack - https://html-agility-pack.net
+            var htmlDoc = new HtmlDocument();
+            htmlDoc.LoadHtml(html);
+
+            var tmpImages = htmlDoc.DocumentNode.SelectNodes("//img[@data-tmpimg]");
+            if (tmpImages == null || tmpImages.Count == 0)
+                return html;
+            
+            foreach (var img in tmpImages)
+            {
+                // The data attribute contains the path to the tmp img to persist as a media item
+                var tmpImgPath = img.GetAttributeValue("data-tmpimg", string.Empty);
```

Comment:
`"data-tmpimg"` could be saved in a local constant.. It is used 3 times in this method. If we wan't to change the name, we should update all 3 locations anyway.